lib/form/deposit.js

```diff
@@ -52,70 +52,51 @@ export default class DepositForm {
             const depositFeePercent = depositFee.toNumber() / 10000;
-
-            const assets = window._tpayment.configs.Assets;
-            for (const asset of Object.keys(assets)) {
-                const bigAmount = this.amountToBig(asset);
-                const discount = await window._tpayment.tchain_contract.getDiscountFee(
-                    assets[asset].Address,
-                    this.discountToken,
-                );
-
-                const discounFeePercent = discount[0].toNumber() / 10000;
-                this._feeDiscounts[asset] = {
-                    percent: discounFeePercent,
-                    deductAmount: discount[1],
-                };
-
-                const rate = this.calcExchangeRate(asset, this._currency, this._rates);
+            const tokenPayFees = window._tpayment.configs.AssetPayFees || [];
+            const tokens = window._tpayment.configs.Assets;
+            for (const token of Object.keys(tokens)) {
+                const rate = this.calcExchangeRate(token, this._currency, this._rates);
                 if (rate === 0) continue;
+                for(let i = 0; i < tokenPayFees.length; i++) {
+                    const discountAsset = tokenPayFees[i];
+                    const discount = await window._tpayment.tchain_contract.getDiscountFee(
+                        tokens[token].Address,
+                        tokens[discountAsset].Address,
+                    );
+
+                    const discounFeePercent = discount[0].toNumber() / 10000;
+                    this._feeDiscounts[token] = {
+                        ...this._feeDiscounts[token],
+                        [discountAsset]: {
+                            percent: discounFeePercent,
+                            deductAmount: discount[1],
+                            deductUnitAmount: ethers.utils.formatUnits(discount[1], 18)
+                        },
+                    };
+                };
 
                 const fee = parseFloat((this._amount / rate) * depositFeePercent).toFixed(18);
                 const feeBig = ethers.utils.parseUnits(fee.toString());
-                this._serviceFee[asset] = {
+                this._serviceFee[token] = {
                     fee: feeBig,
                     totalFee: feeBig,
                     percent: depositFeePercent,
+                    feeDiscount: 0,
                 };
 
-                if (this._isUseToko) {
-                    const feeDiscount = parseFloat((this._amount / rate) * discounFeePercent).toFixed(18);
+                if (this._isUseDiscountToken && this._discountToken !== "" && this._feeDiscounts[token]) {
+                    const discountInfo = this._feeDiscounts[token][this._discountToken];
+                    const feeDiscount = parseFloat((this._amount / rate) * discountInfo.percent).toFixed(18);
                     const feeDiscountBig = ethers.utils.parseUnits(feeDiscount.toString());
```

Comment:
This may need to be refactored later as calculating with float is not precise:
- First convert the amount to big number by using ethers
- Then use ethers' utilities to calculate big amount
- Finally, convert the final amount to float